rts/System/Platform/Linux/CpuTopology.cpp

```diff
@@ -31,9 +31,61 @@ Vendor detect_cpu_vendor() {
 	return VENDOR_UNKNOWN;
 }
 
-// Get number of logical CPUs
-int get_cpu_count() {
-	return sysconf(_SC_NPROCESSORS_CONF);
+std::vector<int> get_online_cpus() {
+	std::vector<int> cpus;
+	std::ifstream file("/sys/devices/system/cpu/online");
+	if (file) {
+		// This is a comma-seperated list of ranges
+		// or single values.
+		// Ex: 0,2,4,6 or 0-7 or 0-3,8-15
+		std::string line;
+		std::getline(file, line);
+		std::istringstream ss(line);
+		int min_cpu;
+		int max_cpu;
+		char sep;
+		while (ss >> min_cpu) {
```

Comment:
why the `ss` var and not just
```suggestion
		while (file >> min_cpu) {
```

Reply:
I copied the pattern from `get_thread_siblings()` in this same file. I guess the idea is that `std::getline()` reads the entire file into memory at once.

I can try it the other way if you like.